pkg/storage/etcd3/watcher.go

```diff
@@ -324,25 +331,40 @@ func (wc *watchChan) sendEvent(e *event) {
 	}
 }
 
-func prepareObjs(ctx context.Context, e *event, client *clientv3.Client, codec runtime.Codec, versioner storage.Versioner) (curObj runtime.Object, oldObj runtime.Object, err error) {
+func prepareObjs(
+	ctx context.Context, e *event, client *clientv3.Client, codec runtime.Codec,
+	versioner storage.Versioner, isEverythingFilter bool) (curObj runtime.Object, oldObj runtime.Object, err error) {
 	if !e.isDeleted {
 		curObj, err = decodeObj(codec, versioner, e.value, e.rev)
 		if err != nil {
 			return nil, nil, err
 		}
 	}
 	if e.isDeleted || !e.isCreated {
-		getResp, err := client.Get(ctx, e.key, clientv3.WithRev(e.rev-1))
```

Comment:
instead of trying to completely remove get, probably we should try to add clientv3.WithSerializable option first. We already specify the revision here. So it is equivalent to the current read, but much cheaper (wont go through raft + hit disk). 

If it does not work, we will try prevKV and back-port it to 3.0 branch this week probably.